### PR TITLE
Detailed of vaccination implemented in another VC.

### PR DESCRIPTION
**Pull request by laxmankhanal**

### CODE REVIEW

File: Vaccination/View/VaccinationCell.swift [hunk 1]
```diff
@@ -13,6 +13,7 @@ class VaccinationCell: UITableViewCell {
     @IBOutlet var alertStrip: UIView!
     @IBOutlet var checkedButton: UIButton!
 
+    var cellSwitch = UISwitch()
```
Comment (sanjaymhj) [inline Vaccination/View/VaccinationCell.swift]: delete this

File: Vaccination/ViewController/AddChildFormTVC.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ class AddChildFormTVC: UITableViewController, UITextFieldDelegate {
         let alert = UIAlertController(title: "Alert", message: "Discard information and exit", preferredStyle: UIAlertControllerStyle.Alert)
         alert.addAction(UIAlertAction(title: "Cancel", style: .Cancel, handler: nil))
         alert.addAction(UIAlertAction(title: "Done", style: .Default, handler: { action in
-            self.dismissViewControllerAnimated(false, completion: nil)
+            self.dismissViewControllerAnimated(true, completion: nil)
```
Comment (sanjaymhj) [inline Vaccination/ViewController/AddChildFormTVC.swift]: should animation set to false

File: Vaccination/ViewController/DetailedVC.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+class DetailedVC: UIViewController {
+
+    @IBOutlet var vaccineNameLabel: UILabel!
+    @IBOutlet var vaccineDoseLabel: UILabel!
+    @IBOutlet var vaccineDescription: UILabel!
+    var childVaccination: ChildVaccination!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        print(childVaccination.vaccination?.name)
```
Comment (sanjaymhj) [inline Vaccination/ViewController/DetailedVC.swift]: delete print

File: Vaccination/ViewController/DetailedVC.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        showData()
```
Comment (sanjaymhj) [inline Vaccination/ViewController/DetailedVC.swift]: better name function as populateVaccinationData

File: Vaccination/ViewController/DetailedVC.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    @IBAction func backButton(sender: UIButton) {
+        self.dismissViewControllerAnimated(true, completion: nil)
+    }
+
+    func showData() {
+        vaccineNameLabel.text = childVaccination.vaccination?.name
+        vaccineDescription.text = childVaccination.vaccination?.vacciDescription
```
Comment (sanjaymhj) [inline Vaccination/ViewController/DetailedVC.swift]: also add the dosage and time after birth to show to the user

File: Vaccination/ViewController/MainVC.swift [hunk 6]
```diff
@@ -319,6 +306,7 @@ extension MainVC: UITableViewDataSource {
             cell.dateLabel.text = String(getDate(withDate: childVaccination.deuDate!))
             cell.monthLabel.text = String(getMonthName(withDate: childVaccination.deuDate!))
             cell.vaccinationName.text = (childVaccination.vaccination!).name
+            cell.cellSwitch.tag = indexPath.row
```
Comment (sanjaymhj) [inline Vaccination/ViewController/MainVC.swift]: do we need tag for the cell?

File: Vaccination/ViewController/MainVC.swift [hunk 7]
```diff
@@ -411,12 +399,15 @@ extension MainVC: UICollectionViewDelegate {
 
 extension MainVC: VaccinationCellDelegate {
     func cellForCheckBoxClicked(cell: VaccinationCell) {
+        let indexPath = NSIndexPath(forRow: cell.tag, inSection: 0)
+        print(indexPath.row)
```
Comment (sanjaymhj) [inline Vaccination/ViewController/MainVC.swift]: remove print

File: Vaccination/ViewController/AddChildFormTVC.swift [hunk 8]
```diff
@@ -38,9 +38,11 @@ class AddChildFormTVC: UITableViewController, UITextFieldDelegate {
         var error = validateChildFormData(withFirstName: self.firstNameTextField.text!, withLastName: self.lastNameTextField.text!)
         if error.count != 0 {
             if error["firstName"] != nil {
```
Comment (anmol-lf) [inline Vaccination/ViewController/AddChildFormTVC.swift]: in scenarios like this where we have to use error["firstName"] inside the block, it'd be better to conditionally assign it to a variable and use that inside

if let firstName = error["firstName"] {
    ....
    ... NSAttributedString(string: firstName ...
    ....
}

File: Vaccination/ViewController/DetailedVC.swift [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
```
Comment (anmol-lf) [inline Vaccination/ViewController/DetailedVC.swift]: add 'self' in lines 12, 13, 21, 22

File: Vaccination/View/VaccinationCell.swift (hunk 1 above)
Comment (laxmankhanal) [inline Vaccination/View/VaccinationCell.swift]: :ok:

File: Vaccination/ViewController/AddChildFormTVC.swift (hunk 2 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/AddChildFormTVC.swift]: :ok:

File: Vaccination/ViewController/DetailedVC.swift (hunk 3 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/DetailedVC.swift]: :ok:

File: Vaccination/ViewController/DetailedVC.swift (hunk 4 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/DetailedVC.swift]: :ok:

File: Vaccination/ViewController/MainVC.swift (hunk 6 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/MainVC.swift]: No it was just to check.

File: Vaccination/ViewController/MainVC.swift (hunk 7 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/MainVC.swift]: :ok:

File: Vaccination/ViewController/AddChildFormTVC.swift (hunk 8 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/AddChildFormTVC.swift]: :ok:

File: Vaccination/ViewController/DetailedVC.swift (hunk 9 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/DetailedVC.swift]: 🆗

File: Vaccination/ViewController/DetailedVC.swift (hunk 5 above)
Comment (laxmankhanal) [inline Vaccination/ViewController/DetailedVC.swift]: 🆗

File: Vaccination/ViewController/EditChildFormTVC.swift [hunk 10]
```diff
@@ -88,19 +90,23 @@ class EditChildFormTVC: UITableViewController {
                 return UITableViewCell()
             }
             if indexPath.row == 0 {
-                nameCell.textField.placeholder = "Enter First Name"
+                self.firstNameTextField = nameCell.textField
+                self.firstNameTextField.text = child!.firstName
+                self.firstNameTextField.placeholder = "Enter First Name"
```
Comment (sanjaymhj) [inline Vaccination/ViewController/EditChildFormTVC.swift]: Why not place it in the seperate constants file

Comment (laxmankhanal) [inline Vaccination/ViewController/EditChildFormTVC.swift]: it comes in code refactoring process

**Comment by sanjaymhj** [comment]
:+1:

File: Vaccination/ViewController/EditChildFormTVC.swift [hunk 11]
```diff
@@ -23,32 +23,34 @@ class EditChildFormTVC: UITableViewController {
...
-        //TODO: Make the changes to database according to user input.
-        child?.firstName = self.firstNameTextField.text
-        child?.lastName = self.lastNameTextField.text
-        child?.gender = self.genderSelected.nameString()
-        child?.setPhoto(withPhoto: self.genderSelected.getDefaultImage())
-        CoreDataHelper.sharedInstance.saveMainContext()
-        cancelEditing(sender)
+        let alert = UIAlertController(title: "Alert", message: "Are you sure to edit information", preferredStyle: UIAlertControllerStyle.Alert)
+        alert.addAction(UIAlertAction(title: "No", style: .Cancel, handler: nil))
+        alert.addAction(UIAlertAction(title: "Yes", style: .Default, handler: { action in
```
Comment (bishalg) [inline Vaccination/ViewController/EditChildFormTVC.swift]: try to use NSLocalized String when ever there is use of displaying String

File: Vaccination/ViewController/EditChildFormTVC.swift [hunk 12]
```diff
@@ -88,19 +90,23 @@ class EditChildFormTVC: UITableViewController {
...
+                self.firstNameTextField.placeholder = "Enter First Name"
             } else {
-                nameCell.textField.placeholder = "Enter Last Name"
+                self.lastNameTextField = nameCell.textField
+                self.lastNameTextField.text = child!.lastName
+                self.lastNameTextField.placeholder = "Enter Last Name"
             }
             return nameCell
 
         case 1:
```
Comment (bishalg) [inline Vaccination/ViewController/EditChildFormTVC.swift]: when-ever there is Switch Case remember we can always have an enum to switch case on enum value rather then 0, 1or 2.

File: Vaccination/ViewController/MainVC.swift [hunk 13]
```diff
@@ -273,20 +271,11 @@ class MainVC: UIViewController {
...
-                } else {
-                    selectedIndexPath = indexPath
-                }
-            }
-            tableView.reloadRowsAtIndexPaths(
-                [indexPath], withRowAnimation: UITableViewRowAnimation.Fade)
-        }
+        let sb = UIStoryboard(name: "Main", bundle: nil)
+        let detailedVC = sb.instantiateViewControllerWithIdentifier("DetailedVCID") as! DetailedVC
+        detailedVC.childVaccination = tableData[indexPath.row]
```
Comment (bishalg) [inline Vaccination/ViewController/MainVC.swift]: use Storyboard constant file to reference Storyboard rather then using it in VC

**Pull request merged**